src/grpc.js

```diff
@@ -307,6 +337,25 @@ class LndGrpc extends EventEmitter {
     }
   }
 
+  /**
+   * Wait for wallet  to enter a particular state.
+   * @param  {string} state state to wait for (RPC_ACTIVE, LOCKED, UNLOCKED)
+   * @return {Promise<Object>}.
+   */
+  checkWalletState(state) {
+    const waitForState = resolve => {
```

Comment:
Maybe we should `reject` after `CONNECT_WAIT_TIMEOUT` otherwise this could hang indefinitely in some edge cases.

Reply:
@mrfelton did the changes, once you review, will clean commits and make ready for merge.